shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs

```diff
@@ -154,26 +155,73 @@ instance CC.Crypto crypto => Val (Value crypto) where
   modifyCoin f (Value c m) = Value n m where (Coin n) = f (Coin c)
   pointwise p (Value c x) (Value d y) = (p c d) && (pointWise (pointWise p) x y)
 
-  size (Value _ v) =
-    -- add uint for the Coin portion in this size calculation
-    foldr accum uint v
-    where
-      -- add addrHashLen for each Policy ID
-      accum u ans = foldr accumIns (ans + policyIdLen) u
-        where
-          -- add assetNameLen and uint for each asset of that Policy ID
-          accumIns _ ans1 = ans1 + assetNameLen + uint
-      -- TODO move these constants somewhere (they are also specified in CDDL)
-      uint :: Integer
-      uint = 9
-
-      assetNameLen :: Integer
-      assetNameLen = 32
-
-      -- TODO dig up these constraints from Era
-      -- address hash length is always same as Policy ID length
-      policyIdLen :: Integer
-      policyIdLen = 28
+-- CompactValueAdaOnly - 2 words
+-- CompactValueMultiAsset
+--  for each distinct asset we have 1.5 words (round up)
+--   4 -- heap object  (3 fields)
+--   2 ShortByteString
+--   size of asset names + pids
+
+  -- = CompactValueAdaOnly {-# UNPACK #-} !Word64
+  -- | CompactValueMultiAsset
```

Comment:
is this scratch work?

Reply:
Yes - sort of, but also I wanted to put a bit of code to explain why the size calculation is what it is. Should I clean it up? Should I remove all commented out code and just keep the description?

Reply:
comments are great, i just couldn't tell if that was outdated code or what it was exactly